ir/value.cpp

```diff
@@ -220,25 +206,22 @@ StateValue Input::toSMT(State &s) const {
     val = expr::mkIf(type.extract(0, 0) == 0, val, undef);
   }
 
-  if (has_deref || has_nonnull) {
+  if (has_deref) {
     Pointer p(s.getMemory(), val);
-    if (has_deref) {
-      s.addAxiom(type == 0);
-      s.addAxiom(p.isDereferenceable(attrs.derefBytes, bits_byte/8, false));
-    }
-    if (has_nonnull && !has_deref) {
-      s.addAxiom(type.extract(1, 1) == 0);
-    }
+    s.addAxiom(p.isDereferenceable(attrs.derefBytes, bits_byte/8, false));
```

Comment:
This neatens the poison encoding of nonnull attribute; `has_nonnull` is added to `never_poison`.

Reply:
Also, since `has_deref` already existed at `never_undef`, `s.addAxiom(type == 0);` is not necessary.